### PR TITLE
Upgrade RandomX to v1.1.7

### PR DESCRIPTION
**Pull request by ldmberman**
v1.1.7 is the latest version of RandomX, which contains major performance improvements (up to 20%).

The commit includes a test for the RandomX compatibility, which can be used in the future
compatible upgrades.

bulk_hash_fast_nif was switched to the new RandomX API, functions calculate_hash_first,
calculate_hash_next, and calculate_hash_last, which provide the main perfomance increase.

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -44,13 +44,13 @@
 ]}.
 
 {pre_hooks, [
-	{"(linux)", compile, "env AR=gcc-ar make -C apps/arweave/lib/RandomX"},
-	{"(darwin)", compile, "make -C apps/arweave/lib/RandomX"},
+	{"(linux|darwin)", compile, "bash -c \"mkdir -p apps/arweave/lib/RandomX/build && cd apps/arweave/lib/RandomX/build && cmake .. > /dev/null\""},
+	{"(linux|darwin)", compile, "make -C apps/arweave/lib/RandomX/build"},
 	{"(linux)", compile, "env AR=gcc-ar make -C apps/arweave/c_src"},
```
Comment (jespern) [inline rebar.config:49]: Does this ever get matched now?

Comment (ldmberman) [reply rebar.config:49]: All steps are executed independently from whether any of the previous steps matched. In this case 2 steps are executed on both linux and darwin, and the third step is different for these architectures.

**Comment by ldmberman** [comment]
[arweave-2.0.0.4.tar.gz](https://github.com/ArweaveTeam/arweave/files/4799529/arweave-2.0.0.4.tar.gz)

**Comment by ldmberman** [comment]
[arweave-2.1.0.1.tar.gz](https://github.com/ArweaveTeam/arweave/files/4885231/arweave-2.1.0.1.tar.gz)

**Pull request merged**